ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -236,6 +308,11 @@ private TransactionSelectionResult evaluateTransaction(final Transaction transac
       LOG.trace("Selected {} for block creation", transaction);
       updateTransactionResultTracking(transaction, effectiveResult);
     } else {
+      final var isIncorrectNonce = isIncorrectNonce(effectiveResult.getValidationResult());
+      if (!isIncorrectNonce || reportFutureNonceTransactionsAsInvalid) {
```

Comment:
this double negative is confusing - if the nonce is not incorrect, the tx gets added to the invalid list?

Reply:
ah so at this point the tx is invalid, and we're saying the problem is not the nonce

Reply:
The intention is to report the following scenarios:
- `isIncorrectNone && reportFutureNonceTransactionsAsInvalid` 
- any other error 

Maybe this could be adapted to 
`if ((reportFutureNonceTransactionsAsInvalid && isIncorrectNone) || !isIncorrectNone)` for better readability?